cachi2/core/config.py

```diff
@@ -25,6 +25,8 @@ class Config(BaseModel, extra="forbid"):
     requests_timeout: int = 300
     concurrency_limit: int = 5
 
+    allow_yarnberry_processing: bool = True
```

Comment:
Why do we need this option ?

Reply:
Our config file needs some love, we kinda managed to get rid of some Go stuff with the hope of getting rid of some more. If we want to add a backend-related option, then we should scope this by a settings group (an INI section if you will) before we roll this out, because then it's going to be harder to get rid of the option. The config change doesn't necessarily need come with this PR and can be introduced in a follow-up PR with more config changes/cleanups.

Reply:
Moreover, some/many (definitely not all, that's a rare exhibit out there) of the config options (in general) could have a CLI counterpart to make behaviour configuration of cachi2 more convenient.

Reply:
> Why do we need this option ?

I added it basing on https://github.com/containerbuildsystem/cachi2/pull/743#issuecomment-2532884713 ,
the comment also explicitly mentions a config option and not a flag.

> Our config file needs some love
Absolutely! However on L17 I see `goproxy_url`, on L19 -- `gomod_download_max_tries`. The config already looks like a flat space for random variables as it is, do you think it is in scope of this change to start reworking it?

Reply:
> Moreover, some/many (definitely not all, that's a rare exhibit out there) of the config options (in general) could have a CLI counterpart to make behaviour configuration of cachi2 more convenient.

This use case makes me think that we might want to have a set of config options that are only configurable by modifying the config file itself, so that indirect users of Cachi2 as part of a backend system couldn't change them.